python/doc/theory/meta_modeling/cross_validation.rst

```diff
@@ -1,186 +1,658 @@
+This is done using a criteria which takes into account for the
+number of coefficients compared to the size of the
+experimental design.
+The corrected leave-one-out error is ([chapelle2002]_, [blatman2009]_ eq. 4.38 page 86):
+
+.. math::
+    \widehat{\operatorname{MSE}}_{LOO}^{*}
+    = \widehat{\operatorname{MSE}}_{LOO} T(n_x, n)
+
+where the penalty factor is:
+
+.. math::
+    T(n_x, n)
+    = \frac{n}{n - (n_x + 1)}  \left(1 + \frac{\operatorname{Tr} \left( \boldsymbol{C}_{emp}^{-1}  \right) }{n} \right)
+
+where :math:`\boldsymbol{C}_{emp}` is the matrix:
 
 .. math::
-   Err^{(i)}  = \dfrac{1}{ |\cX_i|}  \sum\limits_{\underline{x}^{(j)} \in \cX_i} \left[ h(\underline{x}^{(j)}) - \widehat{h}^{(-\cX_i)} {(\underline{x}^{(j)})} \right]^2
 
-in which
-:math:`h(\underline{x}^{(j)}) - \widehat{h}^{(-\cX_i)} {(\underline{x}^{(j)})}`
-is the *predicted residual* defined as the difference between the
-evaluation of :math:`h(\cdot)` and the prediction with
-:math:`\widehat{h}^{(-\cX_i)}(\cdot)` at :math:`\underline{x}^{(j)}`
-in the sub-sample :math:`\cX_i` whose cardinality is :math:`|\cX_i|`.
+    \boldsymbol{C}_{emp} = \frac{1}{n}\boldsymbol{D}^{\textsf{T}}\boldsymbol{D}
+
+and :math:`\operatorname{Tr}` is the trace operator.
+
+K-fold cross-validation
+~~~~~~~~~~~~~~~~~~~~~~~
 
-The approximation errors :math:`Err^{(i)}` are estimated with each of
-the sub-samples :math:`\cX_i`, :math:`i \in \{ 1, \dots, K\}` being
-used as the validation set whereas the remaining sub-samples
-:math:`\cX \setminus \cX_i` being used for training. Finally the
-:math:`K`-fold cross-validation error estimate is obtained as the
-average:
+In this section, we present the naive K-Fold cross-validation.
+Let :math:`k \in \Nset` be a parameter representing the number of
+splits in the data set.
+The :math:`k`-fold cross-validation technique relies on splitting the
+data set :math:`\set{D}` into :math:`k` mutually disjoint sub-samples
+:math:`\set{D}_1, \dots, \set{D}_k`, called the *folds*.
+The corresponding set of indices:
 
 .. math::
-   Err_{Kfold} = \dfrac{1}{K} \sum\limits_{i=1}^{K} Err^{(i)}
 
-As described above, the :math:`K`-fold error estimate can be obtained
-with a single split of the data :math:`\cX` into :math:`K` folds. It
-is worth noting that one can repeat the cross-validation multiple
-times using different divisions into folds to obtain better Monte
-Carlo estimate. This comes obviously with an additional computational
-cost.
+    \set{S}_1 \; \dot{\cup} \; \cdots \; \dot{\cup} \; \set{S}_k
+    = \{1, ..., n\}
+
+and the corresponding set of input observations is:
+
+.. math::
+
+    \set{D} = \set{D}_1 \; \dot{\cup} \; \cdots \; \dot{\cup} \; \set{D}_k.
+
+The next figure presents this type of cross validation.
+
+.. figure:: ../../_static/kfold_validation.png
+    :align: center
+    :alt: K-Fold validation
+    :width: 75%
+
+    *Figure 2.* K-Fold cross-validation.
+
+The :math:`k` folds are generally chosen so that they are of
```

Comment:
```suggestion
The :math:`k` folds are generally chosen so that they are of
```
```suggestion
The :math:`k` folds are generally chosen to be of
```